pandas/core/internals/construction.py

```diff
@@ -140,9 +141,8 @@ def rec_array_to_mgr(
     fdata = ma.getdata(data)
     if index is None:
         index = _get_names_from_index(fdata)
-        if index is None:
-            index = ibase.default_index(len(data))
-    index = ensure_index(index)
+    else:
+        index = ensure_index(index)
```

Comment:
Should this be left unindented (without `else`)? `_get_names_from_index` doesn't necessarily returns an index?

Reply:
Ah, I see you changed that below ;)